### DISCUSSION TITLE
using p5.js with Next.js ( making a gallery of sketches )

### DISCUSSION THREAD

**Discussion by 0xdhrv** [vercel/next.js#16897] [Help]
I have intermediate knowledge of Next.js and p5.js, I like the overall concept of Next.js and now, I use it to for my web development and, I do sketches in p5, so, I want to make a gallery of my p5.js sketches with Next.js. And, I want help to get started.

I thought of putting my sketches in separate folder 'sketches/' and run a for loop based on the count of sketches in sketches, using `getStaticProps()`. But this doesn't work. Can anyone help me to get started?

**Comment by willianxz** [comment]
I was looking for how to make this relationship ...
I found this example here:
https://github.com/aunnnn/next-p5-template
But from what I see it still seems to be a primitive way of relating the NextJs to P5.js
Visually it is very bad, because it causes interference in the Sketch code of P5.JS ... you have to use several things that together seems to be more of a mixed game ... it is nothing cool.
I hope you can create a better relationship than that, when the two are used together in a project ... I think a better way out so far from what I see would be to separately create P5.JS to host it online and import it as an IFRAME within NextJs ... so there is no conflict problem, because they are separate codes.

Greetings from Brazil!!
